### ISSUE TITLE
interactive Ruby interpreter broken in Ghidra 10.0.3

### ISSUE THREAD

**Issue by goatshriek** [goatshriek/ruby-dragon#9]
The v1.0.0 build of the plugin in Ghidra 10.0.3 may not be able to open an interactive Ruby interpreter, instead throwing a class not found exception.

```
NameError: cannot load Java class org.jruby.ext.readline.Readline
      load_ext at org/jruby/ext/jruby/JRubyUtilLibrary.java:201
        <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/readline.rb:9
       require at org/jruby/RubyKernel.java:974
       require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:83
  <module:IRB> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/input-method.rb:130
        <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb/input-method.rb:15
       require at org/jruby/RubyKernel.java:974
       require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:83
        <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/irb.rb:19
       require at org/jruby/RubyKernel.java:974
       require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:83
        <main> at <script>:1
```

There are two possible workarounds for this. First, the issue does not seem to occur in JDK 11 (it has been confirmed in JDK 16). Alternatively, if changing Java versions is not an option, replacing the `support/launch.properties` file in the Ghidra 10.0.3 installation folder with the version from 10.0.2 also seems to resolve the issue.

An issue will be created with the Ghidra project along with a proposed patch within the next 72 hours.

**Comment by goatshriek** [comment]
An issue NationalSecurityAgency/ghidra#3450 has been submitted to the Ghidra repository describing this problem and a potential solution.

**Comment by goatshriek** [comment]
This is also an issue in ghidra 10.0.4. The warning has been expanded to also show for this version of Ghidra in version 1.0.2 of the RubyDragon plugin.

**Comment by goatshriek** [comment]
This has seemingly resolved in 10.2 releases of Ghidra.
